drivers/storage/portworx/components_test.go

```diff
@@ -7408,6 +7408,54 @@ func TestPrometheusInstall(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, prometheus.Spec.Resources.Requests, cluster.Spec.Monitoring.Prometheus.Resources.Requests)
 	require.Equal(t, prometheus.Spec.SecurityContext, cluster.Spec.Monitoring.Prometheus.SecurityContext)
+
+	// Modify replicas, retention, retentionSize, storage, volumes, volumeMounts
+	replicas := int32(2)
+	fakeSCName := "fake-test-sc"
+	cluster.Spec.Monitoring.Prometheus.Replicas = &replicas
+	cluster.Spec.Monitoring.Prometheus.Retention = "10h"
+	cluster.Spec.Monitoring.Prometheus.RetentionSize = "1048MB"
+	cluster.Spec.Monitoring.Prometheus.Storage = &monitoringv1.StorageSpec{
+		VolumeClaimTemplate: monitoringv1.EmbeddedPersistentVolumeClaim{
+			Spec: v1.PersistentVolumeClaimSpec{
+				StorageClassName: &fakeSCName,
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: resource.MustParse("1G"),
+					},
+				},
+			},
+		},
+	}
+	cluster.Spec.Monitoring.Prometheus.VolumeMounts = []v1.VolumeMount{
+		{
+			Name:      "testVlm",
+			MountPath: "/prometheus/",
+		},
+	}
+	cluster.Spec.Monitoring.Prometheus.Volumes = []v1.Volume{
+		{
+			Name: "testVlm",
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "testPxPvc",
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+	cluster.Spec.Monitoring.Prometheus.SecurityContext = nil
```

Comment:
Should we validate that SecurityContext is nil in the created one as well? Or will it get filled in with defaults?

Reply:
SecurityContext was set to some specific values in a previous unit test. So I set SecurityContext to nil so that the [fallback](https://github.com/libopenstorage/operator/blob/da2dfcbbd7ca1cf6a4e9d8a073ac113e614c7ce2/drivers/storage/portworx/component/prometheus.go#L657) logic will set ```runAsGroup``` and  ```fsGroup```, which is required for opening query file (I just added the error message in the PR description before we adding ```runAsGroup``` and ```fsGroup```)